mongo/integration/gridfs_test.go

```diff
@@ -45,6 +46,66 @@ func TestGridFS(x *testing.T) {
 		findIndex(findCtx, mt, mt.DB.Collection("fs.files"), false, "key", "filename")
 		findIndex(findCtx, mt, mt.DB.Collection("fs.chunks"), true, "key", "files_id")
 	})
+	// should not create a new index if index is numerically the same
+	mt.Run("numericallyEqualIndexes", func(mt *mtest.T) {
+		//add indexes with floats to collections manually
+		res := mt.DB.RunCommand(context.Background(),
+			bson.D{
+				{"createIndexes", "fs.files"},
+				{"indexes", bson.A{
+					bson.D{
+						{"key", bson.D{{"filename", float64(1.0)}, {"uploadDate", float64(1.0)}}},
+						{"name", "filename_1_uploadDate_1"},
+					},
+				}},
+			},
+		)
+		assert.Nil(mt, res.Err(), "createIndexes error: %v", res.Err())
+
+		res = mt.DB.RunCommand(context.Background(),
+			bson.D{
+				{"createIndexes", "fs.chunks"},
+				{"indexes", bson.A{
+					bson.D{
+						{"key", bson.D{{"files_id", float64(1.0)}, {"n", float64(1.0)}}},
+						{"name", "files_id_1_n_1"},
+						{"unique", true},
+					},
+				}},
+			},
+		)
+		assert.Nil(mt, res.Err(), "createIndexes error: %v", res.Err())
+
+		mt.ClearEvents()
+		mt.Run("OpenUploadStream", func(mt *mtest.T) {
+			bucket, err := gridfs.NewBucket(mt.DB)
+			assert.Nil(mt, err, "NewBucket error: %v", err)
+
+			_, err = bucket.OpenUploadStream("filename")
+			assert.Nil(mt, err, "OpenUploadStream error: %v", err)
+
+			mt.FilterStartedEvents(func(evt *event.CommandStartedEvent) bool {
+				return evt.CommandName == "createIndexes"
+			})
+			evt := mt.GetStartedEvent()
+			assert.Nil(mt, evt, "Expected that createIndexes wasn't called: %v", evt)
```

Comment:
Printing out the entire event will probably generate a more complex string than we need because it'll print out every struct field. I don't think we can use the assertion library here if we only want to print a part of the event:

```
if evt != nil {
    t.Fatalf("expected no createIndexes events but got %v", evt.Command)
```

Same suggestion for the test below.